src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1297,6 +1316,7 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
 
     tx = r.value();
     init_tm_tx_reply reply;
+    model::producer_identity rolled_pid = tx.pid;
```

Comment:
can we use `last_pid` to cleanup overwritten producer state here ? It seems that it has exactly the same purpose as the `rolled_pid`

Reply:
As chatted offline, last_pid seems somewhat sketchy  and not updated in all cases, added a clarifying comment.